src/client/components/CollaboratorAccessModal/hooks/useActions.ts

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react'
+
+import { Objects } from '@utils/objects'
+
+import { CollaboratorEditPropertyType } from '@meta/user'
+
+export const useActions = (
+  options: Record<string, string>,
+  sections:
+    | 'none'
+    | 'all'
+    | Record<
+        string,
+        {
+          readonly tableData?: boolean
+          readonly descriptions?: boolean
+        }
+      >
+) => {
+  const [selectedSections, setSelectedSections] = useState(sections)
+
+  const toggleOptions = (permission: CollaboratorEditPropertyType): void => {
+    const enabled =
+      typeof selectedSections !== 'string'
+        ? Object.entries(selectedSections).filter(([_, section]) => section[permission] === true).length <
+          Object.keys(options).length
+        : true
+    const newSelectedSections = typeof selectedSections !== 'string' ? Objects.cloneDeep(selectedSections) : {}
```

Comment:
should we add helper for `typeof selectedSections !== 'string'`? used 9 times

Reply:
or event just common variables. like isAll, isNone ?

Reply:
I'm having a lot of issues from typescript trying to wrap that condition somewhere else. let's skip it for now, I can show you what happens later.